packages/dapp-svelte-wallet/api/src/lib-wallet.js

```diff
@@ -555,17 +554,21 @@ export async function makeWallet({
     const brand = brandMapping.petnameToVal.get(brandPetname);
     const { issuer } = brandTable.getByBrand(brand);
 
-    // IMPORTANT: once wrapped, the original purse should never
-    // be used otherwise the UI state will be out of sync.
-    const doNotUse = await E(issuer).makeEmptyPurse();
+    const purseP = E(issuer).makeEmptyPurse();
+    const [purse, balanceIterable] = await Promise.all([
+      purseP,
+      E(purseP).getCurrentAmountNotifier(),
+    ]);
 
-    const purse = makeObservablePurse(E, doNotUse, () =>
-      updatePursesState(purseMapping.valToPetname.get(purse), purse),
-    );
+    // Just notice the balance updates for the purse.
+    observeIteration(balanceIterable, {
+      updateState(_balance) {
+        updatePursesState(purseMapping.valToPetname.get(purse), purse);
+      },
+    });
```

Comment:
Nice!

Reply:
Interestingly, when the `balanceIterable` was a presence, this observeIteration would fail (because it wasn't an actual asyncIterable).  I learned the hard way always to include a `fail` handler.

Would be nice if types could catch this, but I don't see a simple way to do that.